### ISSUE TITLE
Error while trying to configure ArangoDB replication

### ISSUE THREAD

**Issue by samirski** [arangodb/arangodb#4602]
While trying to configure an async replication on an ArangoDB database (using the document https://docs.arangodb.com/3.3/Manual/Administration/Replication/Asynchronous/Components.html) I got this error:
```
JavaScript exception in file 
'/usr/share/arangodb3/js/client/modules/@arangodb/replication.js' at 209,7: 
ArangoError 1470: replication API is not supported on a coordinator
!      throw err;
!      ^
stacktrace: ArangoError: replication API is not supported on a coordinator
    at Object.exports.checkRequestResult 
(/usr/share/arangodb3/js/client/modules/@arangodb/arangosh.js:96:21)
    at waitForResult 
(/usr/share/arangodb3/js/client/modules/@arangodb/replication.js:207:16)
    at setup 
(/usr/share/arangodb3/js/client/modules/@arangodb/replication.js:310:10)
    at Object.setupReplication 
(/usr/share/arangodb3/js/client/modules/@arangodb/replication.js:313:51)
    at <shell command>:1:34
```
The statement I've executed in arangosh is this:
```
db._useDatabase("_system");
require("@arangodb/replication").setupReplication({
  endpoint: "tcp://my.master.server:8529",
  username: "root",
  password: "password",
  verbose: false,
  includeSystem: false,
  incremental: true,
  autoResync: true
});
```
Tried to search the web for a similar issue, but could not. Any idea on what this means and how to enable the replication? 

## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [ ] 3.0
- [ ] 3.1
- [ ] 3.2
- [ x] 3.3.3
- [ ] self-compiled devel branch

Mode:
- [ x] Cluster
- [ ] Single-Server

Storage-Engine:
- [ ] mmfiles
- [ x] rocksdb

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [ ] Linux 
  - [ ] Debian .deb
  - [ x] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [ ] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:

### this is an AQL-related issue:
[ ] I'm using graph features

I'm issuing AQL via:
- [ x] web interface with this browser:     running on this OS:
- [ x] arangosh
- [ ] this Driver:

**Comment by OmarAyo** [comment]
Hi @samirski,

Thanks for opening this ticket and trying ArangoDB.

It seems that you are making some confusion between an "ArangoDB Cluster" and a "Master/Slave" environment.

In particular, the command you are running (setupReplication) is typically used to start asynchronous replication in a Master/Slave environment.

By reading the reported error message (ArangoError 1470: replication API is not supported on a coordinator) it seems you are trying to execute that command on a "Coordinator", which is a component of the "ArangoDB Cluster". 
On an ArangoDB Cluster the replication is synchronous, and all you have to do, after you have deployed the Cluster, is to set _ReplicationFactor_ when you create your _collections_

If you try to run the same command on a single-server (_slave_) it should work (and start replication from a _master_) - but that is not needed on a Cluster

Please let me know if this helps, or if you have additional questions

Thanks,

**Comment by samirski** [comment]
Hi @OmarAyo 

Thank you for your reply. Yes you are right, I have at this moment some difficulties grasping the concept of a cluster environment in ArangoDB. I even posted a question on Stack Overflow (https://stackoverflow.com/questions/48797150/clustering-and-replication-in-arangodb/48803310). So the only way right now to failover is to have a cluster, right? And in that case why would you guys don't allow to set the replication on a database level instead of the collection level? In case of failover you would expect the whole database to be there, not just some collections from it. Or I'm getting it wrong still?

Thanks

**Comment by OmarAyo** [comment]
Hi @samirski,

We are working to improve our documentation to make this more easily for our new users.

Actually there are two "modes" in which you can run ArangoDB that includes automatic failover. One is the already mentioned "ArangoDB Cluster" and the other one is what we call "Active Failover" (or Active/Passive) which was released in 3.3

You have to notice that they are two different deployments:

- "ArangoDB Cluster" is a highly distributed system which uses synchronous replication within its _DBSevers_, and shards and replicates data according to the  _numberOfShards_ and _replicationFactor_ properties that are defined at collection level.

- "Active Failover" is a system which has one or more "Followers" (or _Slaves_) that replicates asynchronously from a "Leader" (or _Master_). In the "Active Failover" setup, the replication is global in the sense that it affects all the databases of the "Leader", and within the same database, all its collections. _numberOfShards_ and _replicationFactor_ properties are not available in such "Active Failover" mode

As stated above, _numberOfShards_ and _replicationFactor_ properties, in the Cluster, are per collection. I will discuss internally if it is possible, as a product enhancement, to have them defined at database level, and have them inherited by default from all collections of that database.

**Comment by samirski** [comment]
@OmarAyo thank you for an extensive answer!
